### PR TITLE
trivial: Update broken mailing list links to new system (postorius)

### PR DESCRIPTION
**Pull request by dbscoach**
I hope I followed contribution guidelines correctly, let me know if I need to change anything.

### CODE REVIEW

File: processes/docs-contributing.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ The sort of contributions that we are looking for are:
 
 For reporting issues about a particular project, please use the relevant repository's issue tracker or post on the [Devel mailing list][mailing-list].
 
-[mailing-list]: https://sel4.systems/lists/listinfo/
+[mailing-list]: https://lists.sel4.systems/postorius/lists/
```
Comment (axel-h) [inline processes/docs-contributing.md:33]: This is a bit out of the scope of this PR, but I wonder if we should point people directly to the `devel` list here? Currently, the other list available are `RFC` and `announce`, which are a bit more special. Seem we will not have more list soon as the traffic is still low.

```suggestion
[mailing-list]: https://lists.sel4.systems/postorius/lists/devel.sel4.systems/
```

Aside from this, we could also recommend the Discourse forums at https://sel4.discourse.group/...

**Comment by axel-h** [comment]
Thanks for fixing the links. Looks ok to me and good to be to be merged once the GitLint check also passes.

**Comment by dbscoach** [comment]
Sorry about that, I hope it's correct now but I'll have to fix it tomorrow if it isn't.
Cheers from Germany where it's midnight.

**Review by axel-h — APPROVED** [review]

File: processes/docs-contributing.md (hunk 1 above)
Comment (lsf37) [reply processes/docs-contributing.md:33]: Yes, the sentence specifically mentions the Devel list, so updating the link to that would make sense. Let's do that separately.

**Review by lsf37 — APPROVED** [review]

**Comment by lsf37** [comment]
> Sorry about that, I hope it's correct now but I'll have to fix it tomorrow if it isn't.
> Cheers from Germany where it's midnight.

No worries, worked all fine. Thanks for the contribution!

**Pull request merged**